src/session/session_compact.c

```diff
@@ -334,6 +334,32 @@ __wt_session_compact(WT_SESSION *wt_session, const char *uri, const char *config
     session = (WT_SESSION_IMPL *)wt_session;
     SESSION_API_CALL(session, compact, config, cfg);
 
+    /* Trigger the background server thread and toggle connection statistic. */
+    if ((ret = __wt_config_getones(session, config, "background", &cval) == 0)) {
+        if (uri != NULL)
+            WT_ERR_MSG(session, EINVAL, "Background compaction does not work on specific URIs.");
+
+        if (!cval.val) {
```

Comment:
Can we add a comment here saying we know background compaction has been explicitly disabled and we want to ensure no other parameters have been given?